pyjexl/jexl.py

```diff
@@ -71,19 +72,20 @@ def wrapper(func):
     def parse(self, expression):
         try:
             return Parser(self.config).visit(self.grammar.parse(expression))
-        except ParsimoniousParseError as err:
-            raise ParseError('Could not parse expression: ' + expression) from err
+        except ParsimoniousParseError, err:
+            raise ParseError('Could not parse expression: ' + expression)
 
     def analyze(self, expression, AnalyzerClass):
         parsed_expression = self.parse(expression)
         visitor = AnalyzerClass(self.config)
         return visitor.visit(parsed_expression)
 
+
     def validate(self, expression):
         try:
-            yield from self.analyze(expression, ValidatingAnalyzer)
-        except ParseError as err:
-            yield str(err)
+            for res in self.analyze(expression, ValidatingAnalyzer): yield res
+        except ParseError, err:
+            yield unicode(err)
```

Comment:
`unicode` doesn't exist in Python 3. Since you added `future`, we can [import str to fix this instead](http://python-future.org/compatible_idioms.html#unicode).